operations/helm/charts/alloy/values.yaml

```diff
@@ -207,8 +207,10 @@ controller:
   enableStatefulSetAutoDeletePVC: false
 
   autoscaling:
-    # -- Creates a HorizontalPodAutoscaler for controller type deployment.
+    # -- Creates an autoscaler for the controller.
     enabled: false
+    # -- Type of autoscaler to use for the controller. Must be one of 'vertical' or 'horizontal'.
+    flavor: horizontal
```

Comment:
Do we want to use the `flavor` option, rather than `enable` flags for either option?
This would make it harder to support HPAs and VPAs at the same time if we chose to implement the logic to do so.

Reply:
To be fair I wanted to use this to not break existing stuff and it feels weird to have autoscaling and verticalautoscaling next to each other.

ideally, I think this should be changed to something like (this would require a major version though):

```yaml
  autoscaling:
    horizontal:
      enabled: false
    vertical:
      enabled: false
```

We could then envision a flavor flag under horizontal to be kubernetes or keda for instance in the far future.

What do you think?

Reply:
Another idea as to not break the chart, we can do something like?

```
autoscaling: 
  enabled: false
  horizontal:
    enabled: true # This keeps the behavior
  vertical:
    enabled: false
```

I'm fine with implementing both options :)

Reply:
Yeah, I kinda like that one the best.

Reply:
Oh but that does not work with the other values like autoscaling.minReplicas :(

So either we move to something like:

```
autoscaling:
  enabled: false
  minReplicas: 1
verticalPodAutoscaling:
  enabled: false
```

Or we need some kind of breaking change to accomodate the 2. Sadly I'm not able to find a helm chart that supports both that could be used as a reference

Reply:
Ok, so doing some research, it looks like you can only use HPA and VPA at the same time if you make sure you use separate metrics. I.e. don't let the HPA scale replicas based on CPU and the VPA scales the pod resources based on CPU.

At this point, I'm backtracking my original feelings. If it seems that doing it "the right way" would mean a breaking change, especially one that we haven't had a request for yet, then maybe we save that for the future.

I'll defer to your judgement.

Sometime, I'd love to see if we can set up an HPA to scale based on a custom prometheus metric like active series...

Reply:
So you are right that HPA and VPA should not be used together on the same metric mostly because side-effect is unknown. The issue I have is that we run one of our alloy as a DaemonSet (for logs) and HPA by design does not work with DaemonSets so in my opinion, HPAs should be the default choice but users should be able to use VPAs (and maybe both HPA and VPAs) depending on their use case.

For instance, I could see value in VPA for:
- DaemonSets as stated above
- If users want less beefy alloy instances instead of using clustering (because of node pool topologies and so on)

So after some thought, I think it's probably better to have the breaking change now than later and I would propose to have scaling information done like so:

```yaml
autoscaling:
  horizontal:
    enabled: false
    ... other horizontal values
  vertical:
    enabled: false
    ... other horizontal values
```

I'm not really a fan of breaking changes but I'm not sure we can do better :(

If you're okay with it I'm fine implementing this